lib/aruba/cucumber/command.rb

```diff
@@ -28,6 +28,21 @@
   run_simple(cmd, :fail_on_error => true, :exit_timeout => secs && secs.to_i)
 end
 
+When(/I run the following script:$/) do |file_content|
+  prepend_environment_variable('PATH', expand_path('bin') + ':')
+  step 'an executable named "./bin/myscript" with:', file_content
+  step 'I run `myscript`'
+end
+
+When(/^I run the following (bash|zsh|fish|dash)? ?commands?:$/) do |shell_type, file_content|
```

Comment:
1. Style of regex
  We used the `(?:(bash|zsh|fish|dash) )?` such a thing in other steps as well. Would be great if you make yours look the same. Something similar to this:
  
  ~~~
  When(/^I run the following (?:(bash|zsh|fish|dash) )?commands?:$/)
  ~~~

2. Explicit state the shell
  
  Is it really required (for your use case) to tell what shell is used or can we use `ENV['SHELL']` instead?

  @SteveDonie Do you know if this works on windows too?

3. Extract "generator" for scripts into class(es)

  Given we need to support windows too, I think it might make sense to extract the logic in to a `ScriptGenerator`-classes, maybe even a single class for each shell type or just a single one for windows/unix.
  
  * UnixScriptGenerator
  * WindowsScriptGenerator < UnixScriptGenerator